frontend/src/components/shortener-input/handleShortener.ts

```diff
@@ -90,3 +82,26 @@ export async function handleShortener({ state }: any) {
   copyUrl(state);
   confettiAnimate();
 }
+
+/**
+ * Normalize input url
+ *  - add protocol 'http' if missing.
+ *  - correct protocol http/https if mistyped one character.
+ * @param {String} url
+ * @returns {String}
+ */
+const normalizeUrl = (url: string): string => {
+  const regexBadPrefix = new RegExp(/^(:\/*|\/+|http:\/*)/); // Check if starts with  ':', '/' and 'http:example.com' etc.
+  const regexBadPrefixHttps = new RegExp(/^https:\/*/); // Check if 'https:example.com', 'https:/example.com' etc.
+  const regexProtocolExists = new RegExp(/^(.+:\/\/|[^a-zA-Z])/); // Check if starts with '*://' or special chars.
+  const regexMistypedHttp = new RegExp(
+    /^([^hH][tT][tT][pP]|[hH][^tT][tT][pP]|[hH][tT][^tT][pP]|[hH][tT][tT][^pP])/
+  );
+
+  url = url.replace(regexMistypedHttp, 'http');
+  url = url.replace(regexBadPrefix, 'http://');
+  url = url.replace(regexBadPrefixHttps, 'https://');
```

Comment:
Please make the default protocol to be "https" instead of "http"
Also pay attention that .replace() returns a new string, we can use it like this: (Also changed to https)

```js
url = url.replace(regexMistypedHttp, 'https')
         .replace(regexBadPrefix, 'https://')
         .replace(regexBadPrefixHttps, 'https://');
```

Reply:
if url has http: (not https:), will keep as http:. For example: http:example.com -> http://example.com